apps/sr/tid1500reader.cxx

```diff
@@ -0,0 +1,218 @@
+  CHECK_COND(doc.getCurrentRequestedProcedureEvidence().addItem(*ff.getDataset()));
+}
+
+bool isCompositeEvidence(OFString& sopClassUID) {
+  const char* compositeContextSOPClasses[] = {UID_SegmentationStorage, UID_RealWorldValueMappingStorage};
+  int length = sizeof(compositeContextSOPClasses)/sizeof(*compositeContextSOPClasses);
+  for( unsigned int i=0; i<length; i++)
+    if (sopClassUID == compositeContextSOPClasses[i])
+      return true;
+  return false;
+}
+
+Json::Value getMeasurements(DSRDocument &doc) {
+  Json::Value measurements(Json::arrayValue);
+  DSRDocumentTree &st = doc.getTree();
+
+  DSRDocumentTreeNodeCursor cursor;
+  st.getCursorToRootNode(cursor);
+  if(st.gotoNamedChildNode(CODE_DCM_ImagingMeasurements)) {
+    // can have 1-n MeasurementGroups
+    do {
+      if (st.gotoNamedChildNode(CODE_DCM_MeasurementGroup)) {
+        Json::Value measurement;
+        if (st.gotoNamedChildNode(DSRCodedEntryValue("C67447", "NCIt", "Activity Session"))) {
+          // TODO: think about it
+          cout << "Activity Session: " << st.getCurrentContentItem().getStringValue().c_str() << endl;
+          measurement["ActivitySession"] = st.getCurrentContentItem().getStringValue().c_str();
+        }
+        st.gotoParent();
+        if (st.gotoNamedChildNode(CODE_DCM_ReferencedSegment)) {
+          DSRImageReferenceValue referenceImage = st.getCurrentContentItem().getImageReference();
+          OFVector<Uint16> items;
+          referenceImage.getSegmentList().getItems(items);
+          cout << "Reference Segment: " << items[0] << endl;
+          measurement["ReferencedSegment"] = items[0];
+          if (!referenceImage.getSOPInstanceUID().empty()){
+            measurement["segmentationSOPInstanceUID"] = referenceImage.getSOPInstanceUID().c_str();
+          }
+          st.gotoParent();
+        }
+        st.gotoParent();
+        if (st.gotoNamedChildNode(CODE_DCM_SourceSeriesForSegmentation)) {
+          cout << "SourceSeriesForImageSegmentation: " << st.getCurrentContentItem().getStringValue().c_str() << endl;
+          measurement["SourceSeriesForImageSegmentation"] = st.getCurrentContentItem().getStringValue().c_str();
+          st.gotoParent();
+        }
+        if (st.gotoNamedChildNode(CODE_DCM_TrackingIdentifier)) {
+          cout << "TrackingIdentifier: " << st.getCurrentContentItem().getStringValue().c_str() << endl;
+          measurement["TrackingIdentifier"] = st.getCurrentContentItem().getStringValue().c_str();
+          st.gotoParent();
+        }
+        if (st.gotoNamedChildNode(CODE_DCM_Finding)) {
+          measurement["Finding"]["CodeMeaning"] = st.getCurrentContentItem().getCodeValue().getCodeMeaning().c_str();
+          measurement["Finding"]["CodingSchemeDesignator"] = st.getCurrentContentItem().getCodeValue().getCodingSchemeDesignator().c_str();
+          measurement["Finding"]["CodeValue"] = st.getCurrentContentItem().getCodeValue().getCodeValue().c_str();
```

Comment:
Haha I thought for now (for the RSNA) it's fine.... Refactoring later.... I first want to get everything work....